src/errors.rs

```diff
@@ -15,8 +15,8 @@ pub enum Error {
     Corruption(String),
     #[error("IO Error: {0}")]
     Io(#[from] IoError),
-    #[error("Fsync Error: {0}")]
-    Fsync(String),
+    #[error("Fsync Error: reason: {0}")]
+    Severe(String),
```

Comment:
There is no need for making another error type, if:
1) We panic when fsync failed, no matter the error type is fatal or else
2) We don't propagate the error to multiple writers, so don't need a cloneable error type